### PR TITLE
Add sendmail

### PR DESCRIPTION
**Pull request by immae**
This PR adds support for sendmail along smtp for e-mail configuration.

### CODE REVIEW

File: config/production.yaml.example [hunk 1]
```diff
@@ -53,6 +53,8 @@ redis:
 
 # SMTP server to send emails
 smtp:
+  transport: smtp
```
Comment (Chocobozzz) [inline config/production.yaml.example:56]: Please add these keys in https://github.com/Chocobozzz/PeerTube/blob/develop/config/default.yaml too

File: server/lib/emailer.ts [hunk 2]
```diff
@@ -41,33 +41,41 @@ class Emailer {
...
+          port: CONFIG.SMTP.PORT,
+          secure: CONFIG.SMTP.TLS,
+          debug: CONFIG.LOG.LEVEL === 'debug',
+          logger: bunyanLogger as any,
+          ignoreTLS: CONFIG.SMTP.DISABLE_STARTTLS,
+          tls,
+          auth
+        })
+      } else { // sendmail
+        this.transporter = createTransport({
```
Comment (Chocobozzz) [inline server/lib/emailer.ts:73]: Please add log "Using sendmail to send emails"

File: config/production.yaml.example (hunk 1 above)
Comment (Chocobozzz) [inline config/production.yaml.example:56]: Add a comment `smtp or sendmail`

File: config/production.yaml.example [hunk 3]
```diff
@@ -53,6 +53,8 @@ redis:
...
+  sendmail: null
```
Comment (Chocobozzz) [inline config/production.yaml.example:57]: Add a comment `Path to sendmail command. Required if you use sendmail transport`

**Comment by immae** [comment]
Fixed @Chocobozzz thanks for the review!

**Comment by Chocobozzz** [comment]
Thanks @immae

**Pull request merged**